### PR TITLE
chore: eliminate the use of `xxx.Construct` to reduce v2 merge conflicts

### PR DESCRIPTION
**Pull request by eladb**
Introduce an eslint rule & fix all instances in our code which refer to `@aws-cdk/core.Construct` using a qualified namespace (e.g. `core.Construct`, `cdk.Construct`, etc). In v2, `Construct` will refer to `constructs.Construct` so it will reduce the chances for merge issues.

The rule fixer automatically adds an `import` statement (separated from the main import group) and replaces the usage. If the file already imports `constructs.Construct`, then the new import will alias as `CoreConstruct`.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: tools/cdk-build-tools/bin/cdk-build.ts [hunk 1]
```diff
@@ -29,6 +29,11 @@ async function main() {
       desc: 'execute gen script',
       default: true,
     })
+    .option('fix', {
+      type: 'boolean',
+      desc: 'run eslint with --fix',
```
Comment (nija-at) [inline tools/cdk-build-tools/bin/cdk-build.ts:34]: not just eslint. pkglint and all other linters, right?

File: tools/eslint-plugin-cdk/lib/rules/no-core-construct.ts [hunk 2]
```diff
@@ -9,23 +9,19 @@ const BANNED_TYPES = [ 'IConstruct', 'Construct' ];
 export function create(context: Rule.RuleContext): Rule.NodeListener {
   return {
 
-    // `node` is a type from @typescript-eslint/typescript-estree, but using 'any' for now
-    // since it's incompatible with eslint.Rule namespace. Waiting for better compatibility in
-    // https://github.com/typescript-eslint/typescript-eslint/tree/1765a178e456b152bd48192eb5db7e8541e2adf2/packages/experimental-utils#note
-    // Meanwhile, use a debugger to explore the AST node.
```
Comment (nija-at) [inline tools/eslint-plugin-cdk/lib/rules/no-core-construct.ts:15]: minor: Would prefer to keep this line as a note to future developers on how to work on this linter (using the debugger). This is how I did it at least.

File: packages/@aws-cdk-containers/ecs-service-extensions/lib/environment.ts [hunk 3]
```diff
@@ -3,6 +3,10 @@ import * as ecs from '@aws-cdk/aws-ecs';
 import * as cdk from '@aws-cdk/core';
 import { EnvironmentCapacityType } from './extensions/extension-interfaces';
 
+// keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct } from '@aws-cdk/core';
+
```
Comment (nija-at) [inline packages/@aws-cdk-containers/ecs-service-extensions/lib/environment.ts:9]: Nice! This should significantly reduce merge conflicts.

This is what I had originally suggested here - https://github.com/aws/aws-cdk/pull/12054#pullrequestreview-555364422

**Review by nija-at — APPROVED** [review]

File: tools/cdk-build-tools/bin/cdk-build.ts (hunk 1 above)
Comment (eladb) [reply tools/cdk-build-tools/bin/cdk-build.ts:34]: Done

File: tools/eslint-plugin-cdk/lib/rules/no-core-construct.ts (hunk 2 above)
Comment (eladb) [reply tools/eslint-plugin-cdk/lib/rules/no-core-construct.ts]: done

**Pull request merged**